lessons/219/apache/httpd-mpm.conf

```diff
@@ -2,11 +2,26 @@
     PidFile "logs/httpd.pid"
 </IfModule>
 
+## Since testing on instance with 2vCPU, 
+## ThreadsPerChild increased to 64 (more aggressive for 2 vCPUs)
+## ServerLimit set to 8 (2 vCPUs * 4)
+## MaxRequestWorkers set to 512 (64 * 8)
+## MaxConnectionsPerChild set to 10000 as you suggested (helps prevent memory leaks while still allowing good performance)
 <IfModule mpm_event_module>
-    StartServers             3
-    MinSpareThreads         75
-    MaxSpareThreads        250
-    ThreadsPerChild         25
-    MaxRequestWorkers      400
-    MaxConnectionsPerChild   0
+    StartServers             4
```

Comment:
Just curious if this has any benefit, you will now run 4 processes, while you only have 2 cores (threads). It's not like having 2 literal cores, it's just have two 'threads'. Even with multi-threading the docker container will only have 2 threads to run the process. 

If 4 servers make sense, maybe I should increase the Nginx threads to 4 as well? https://github.com/antonputra/tutorials/pull/326/files#diff-b5388329a6f40563cb48fa67327f1fbe1dd347eb7db44012fa0ebfcf9a0c30baR5

Not sure.

Reply:
4 servers with a max of 8 could actually be detrimental for performance when only using 2 vCPUs.

@antonputra Before you run a 4 or 8 hour test. You might want to briefly check which for both Nginx & Apache2 what the "sweet spot" would be in terms of number of threads/servers. And whether overallocation will benefit performance on Docker or not.